csrc/rsa_gen.cpp

```diff
@@ -29,11 +29,18 @@ JNIEXPORT jlong JNICALL Java_com_amazon_corretto_crypto_provider_RsaGen_generate
         BigNumObj bne;
         jarr2bn(env, pubExp, bne);
 
-        if (1 != RSA_generate_key_ex(r, bits, bne, NULL))
-        {
+        if (1 != RSA_generate_key_ex(r, bits, bne, NULL)) {
             throw_openssl("Unable to generate key");
         }
 
+#ifdef ACCP_FIPS_BUILD
```

Comment:
instead of adding a new build flag, why not use this function to determine FIPSness as we do elsewhere?

https://github.com/aws/aws-lc/blob/ae87faf735c0241a115542b1c1022d125564bf55/include/openssl/crypto.h#L92

Reply:
In the Java part of ACCP, you're right that we check FIPSness during run-time and I guess that's because in Java pre-processing the code is not common; however, in C/C++, we can do this at compile time and have a little bit of performance improvement compared to doing it at run-time.

Reply:
I agree that the performance gain is minuscule, but another benefit is that we would not have that code path when we build for FIPS; having said that, I switched to using `FIPS_mode` API in the latest patch.